src/main/haskell/kore/src/Kore/Exec.hs

```diff
@@ -131,6 +138,29 @@ exec indexedModule strategy purePattern = do
   where
     Valid { patternSort } = extract purePattern
 
+-- | Project the value of the exit cell, if it is present.
+execGetExitCode
+    :: VerifiedModule StepperAttributes Attribute.Axiom
+    -- ^ The main module
+    -> ([Rewrite] -> [Strategy (Prim Rewrite)])
+    -- ^ The strategy to use for execution; see examples in "Kore.Step.Step"
+    -> CommonStepPattern Object
+    -- ^ The final pattern (top cell) to extract the exit code
+    -> Simplifier ExitCode
+execGetExitCode indexedModule strategy' purePattern =
+    case resolveSymbol indexedModule $ noLocationId "LblgetExitCode" of
```

Comment:
Why use a hardcoded symbol name here?

Reply:
It's not defined anywhere else

Reply:
What is LblgetExitCode? Is it a builtin, a function defined by the kore prelude, a function which is sometimes defined? I assume it's the latter.

Anyway, there are two issues here: 
1. It seems to be dependent on the algorithm the frontend uses to generate names; in the other similar cases, i.e. builtin functions, we used attributes to identify them.
2. It seems specific to a given semantics.

I suggest either making an attribute for this, or sending it as a command line argument. If none of these is reasonable, I would define a string constant in this module.

Reply:
This function is always going to be given this name by the k frontend if it exists because that is how it is implemented in the llvm backend. The decision was made so that static source files that are not dynamically generated can be statically linked against this function without the need for dynamically generated glue code. I'm not opposed to an attribute if it makes you feel more comfortable, but it's not specific to a semantics; it's generated by the k compiler as an implementation of the "exit" cell attribute. There is no advantage to making it configurable in one backend but not the other, and it was made this way in the llvm backend deliberately to simplify the design.

Reply:
Ok, I see. Given this context, it makes a bit more sense, although I don't know why we need the static linkage you mentioned. My preference for an attribute was based on:
* we use attributes instead of relying on raw names for builtins, and this seems similar enough;
* an attribute would be within the compiler's control, while the symbol name space belongs to both the k compiler and to the user. Yes, I know that users can use random attributes, but it's expected that some of them are reserved for the compiler, and there are reasonable (IMO) ways to have namespace-like stuff there.

Anyway, given that the LLVM backend already uses this name, I guess it's acceptable to also use it here. I would still define that string constant I mentioned earlier. I wonder if @traiansf has any thoughts about this.

Reply:
For reference, the getExitCode function needs to be called by the main function in the llvm backend. We could of course have simply introduced a dynamic glue function, but this design allows us to simply declare a function prototype for it and call it from the source file for the main function. Which would not be possible if we did not know the name of the function statically.